haystack/nodes/prompt/prompt_template.py

```diff
@@ -241,6 +286,26 @@ def __init__(
     def output_variable(self) -> Optional[str]:
         return self.output_parser.output_variable if self.output_parser else None
 
+    @tenacity.retry(
+        reraise=True,
+        retry=tenacity.retry_if_exception_type((HTTPError, RequestException, JSONDecodeError)),
+        wait=tenacity.wait_exponential(multiplier=PROMPTHUB_BACKOFF),
+        stop=tenacity.stop_after_attempt(PROMPTHUB_MAX_RETRIES),
+    )
+    def _get_prompt_template_from_hub(self, name):
+        """
+        Looks for the given prompt in the PromptHub if the prompt is not in the local cache.
+
+        Raises ValueError if the prom
+        """
+        try:
+            prompt_data: prompthub.Prompt = prompthub.fetch(name, timeout=PROMPTHUB_TIMEOUT)
+        except HTTPError as http_error:
+            if http_error.response.status_code != 404:
+                raise http_error
+            raise PromptNotFoundError(f"Prompt template named '{name}' not available in the Prompt Hub.")
+        return name, prompt_data.text
```

Comment:
Why do we return both the name and the text? The caller context should already have the name. 🤔